router/batchrouter/batchrouter.go

```diff
@@ -1593,18 +1615,34 @@ func (worker *workerT) workerProcess() {
 					IgnoreCustomValFiltersInQuery: true,
 					PayloadSizeLimit:              brt.payloadLimit,
 				}
-				toRetry := brt.jobsDB.GetToRetry(queryParams)
+
+				toRetry, err := jobsdb.QueryJobsResultWithRetries(context.Background(), brt.jobdDBQueryRequestTimeout, brt.jobdDBMaxRetries, func(ctx context.Context) (jobsdb.JobsResult, error) {
+					return brt.jobsDB.GetToRetry(ctx, queryParams)
+				})
+				if err != nil {
+					brt.logger.Errorf("BRT: %s: Error while reading from DB: %v", brt.destType, err)
+					panic(err)
+				}
 				combinedList = toRetry.Jobs
 				if !toRetry.LimitsReached {
 					queryParams.JobsLimit -= len(toRetry.Jobs)
 					if queryParams.PayloadSizeLimit > 0 {
 						queryParams.PayloadSizeLimit -= toRetry.PayloadSize
 					}
-					unprocessed := brt.jobsDB.GetUnprocessed(queryParams)
+					unprocessed, err := jobsdb.QueryJobsResultWithRetries(context.Background(), brt.jobdDBQueryRequestTimeout, brt.jobdDBMaxRetries, func(ctx context.Context) (jobsdb.JobsResult, error) {
+						return brt.jobsDB.GetUnprocessed(ctx, queryParams)
+					})
+					if err != nil {
+						brt.logger.Errorf("BRT: %s: Error while reading from DB: %v", brt.destType, err)
+						panic(err)
+					}
 					combinedList = append(combinedList, unprocessed.Jobs...)
 				}
 				brtQueryStat.End()
 				brt.logger.Debugf("BRT: %s: DB Read Complete for parameter Filters: %v retryList: %v, unprocessedList: %v, total: %v", brt.destType, parameterFilters, len(toRetry.Jobs), len(combinedList)-len(toRetry.Jobs), len(combinedList))
+
+				brtQueryStat.End()
```

Comment:
There is also a `brtQueryStat.End()` in line 1641. 

Calling `.End()` twice without a `.Start` in between can lead to double reporting.

 Also, usage of `.Start` / `.End` is not recommended as it is not safe in concurrent situations. Let's address this by using `.Since` or `.SendTiming`

Reply:
damn, thanks for pointing it out. But, have been added by mistaking while resolving some merge conflict. Removed.